workflow/torch/requires_grad.py

```diff
@@ -1,29 +1,32 @@
-from contextlib import contextmanager
+from functools import wraps, partial
 
 
-@contextmanager
-def requires_grad(module, grad=True):
-    '''
-    Usage:
-    with requires_grad(model):
-        prediction = model(features)
-    '''
-    requires_grads = [
-        p.requires_grad for p in module.parameters()
-    ]
-    for p in module.parameters():
-        p.requires_grad = grad
+class RequiresGrad:
+    '''Can be used as a decorator or context manager'''
+    def __init__(self, module, grad=True):
+        self.module = module
+        self.grad = grad
+        self.requires_grads = [
+            p.requires_grad for p in module.parameters()
+        ]
 
-    yield module
+    def __enter__(self):
+        for p in self.module.parameters():
```

Comment:
Will this not mess up parameters that originally didn't require grad? 
Don't we need to do
```
for p, requires_grad in zip(self.module.parameters(), self.requires_grads):
    p.requires_grad = self.grad and requires_grad
```

Reply:
Example
```
>>> class MyModel(nn.Module):
>>>     def __init__(self):
>>>         super().__init__()
>>>         self.param = nn.Parameter(torch.randn(1), requires_grad=False)
>>>         self.dense = nn.Linear(1, 1)
>>>     def forward(self, x):
>>>         return self.dense(x) + self.param

>>> model = MyModel()
>>> for p in model.parameters():
>>>     print(p.requires_grad)
False
True
True

>>> with requires_grad(model):
>>>     for p in model.parameters():
>>>         print(p.requires_grad)
True
True
True
```
Shouldn't be desired behaviour?